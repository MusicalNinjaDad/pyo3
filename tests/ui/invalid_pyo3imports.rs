#![cfg(all(feature = "testing", not(any(PyPy, GraalPy))))]
use pyo3::prelude::*;

#[pyfunction]
#[pyo3(name = "addone")]
fn py_addone(num: isize) -> isize {
    num + 1
}

#[pymodule]
#[pyo3(name = "adders")]
fn py_adders(module: &Bound<'_, PyModule>) -> PyResult<()> {
    module.add_function(wrap_pyfunction!(py_addone, module)?)?;
    Ok(())
}

#[pyo3test]
#[pyo3import(py_adders: form adders import addone)]
fn typo_first_keyword() {}

#[pyo3test]
#[pyo3import(py_adders: from adders improt addone)]
fn typo_second_keyword() {}

fn main() {}