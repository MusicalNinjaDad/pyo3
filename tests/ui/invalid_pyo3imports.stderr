error: invalid import statement
  --> tests/ui/invalid_pyo3imports.rs:18:25
   |
18 | #[pyo3import(py_adders: form adders import addone)]
   |                         ^^^^

error: invalid import statement
  --> tests/ui/invalid_pyo3imports.rs:22:37
   |
22 | #[pyo3import(py_adders: from adders improt addone)]
   |                                     ^^^^^^

error: expected attribute arguments in parentheses: #[pyo3import(...)]
  --> tests/ui/invalid_pyo3imports.rs:26:3
   |
26 | #[pyo3import]
   |   ^^^^^^^^^^

error: expected `:`
  --> tests/ui/invalid_pyo3imports.rs:30:24
   |
30 | #[pyo3import(py_adders from adders import addone)]
   |                        ^^^^
